src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
```

Comment:
Consider splitting this code into two smaller functions for readability, e.g.:

`function filterByMother(people) {`
  `return people.filter(p => people.some(m => m.name === p.mother));`
`}`

`function filterBySon(people) {`
  `return people.filter(p => p.sex === 'm' && people.some(m => m.name === p.mother));`
`}`

Than can we use these for final filter:

`const children = onlyWithSon ? filterBySon(people) : filterByMother(people);`